aten/src/ATen/native/native_functions.yaml

```diff
@@ -2638,6 +2693,11 @@
 - func: linear.out(Tensor input, Tensor weight, Tensor? bias=None, *, Tensor(a!) out) -> Tensor(a!)
   python_module: nn
 
+- func: mps_linear(Tensor self, Tensor weight, Tensor? bias=None) -> Tensor
```

Comment:
From offline discussions: This might be acceptable if there is non-trivial checks in the general function that you want to get "for free".
There is still the option to move these checks in a helper function in a header and just call it from both functions though!

Reply:
If you do `CompositeImplicitAutograd` and `MPS` as keys. And you have an entry in `derivatives.yaml` for `linear`.
Then the autograd codegen should fill in `AutogradMPS` key with the proper autograd kernel no?

Reply:
oh yep, if we add a derivative entry for `linear` then I think that should work

Reply:
Done.